### PR TITLE
Add runnable examples, fix filetype image upload bug

### PR DESCRIPTION
**Pull request by AbigailWoolley**
**Fix Image Upload Bug and add Runnable Exmaples**

**Summary**:  
This PR does 2 major things:
1. Addresses an issue with image uploads. The image_params dictionary was not in compliance with the documented specifications. This pull request resolves that issue and adds additional tests for exceptional scenarios and mismatches between image_params["image_type_id"] and image_params["file_name"].
2. Adds runnable examples for all products.

### CODE REVIEW

File: examples/smartSelfie_authentication.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+"""SmartSelfie™ Authentication lets you verify an existing a user.
```
Comment (mldangelo) [inline examples/smartSelfie_authentication.py]: can you please make this file and smartSelfie_authentication both snake_case?

**Review by mldangelo — CHANGES_REQUESTED** [review]
One change, everything else looks good. Thanks!

File: examples/smartSelfie_authentication.py (hunk 1 above)
Comment (AbigailWoolley) [reply examples/smartSelfie_authentication.py]: Done.

**Review by mldangelo — APPROVED** [review]
Thanks!

**Pull request merged**